pkg/scheduler/tests/scheduler_recovery_test.go

```diff
@@ -234,7 +236,24 @@ partitions:
 	err = ms.Init(configData, false)
 	assert.NilError(t, err, "2nd RegisterResourceManager failed")
 
-	// Register nodes, and add apps
+	// Register apps first
+	// app must be recovered first in prior of allocations
+	// this is because each allocation will need to have the
+	// app it belongs to registered in the scheduler before the recovery
```

Comment:
That's true. But that can not guarantee the apps are always registered before recovering the existing allocations. I saw sometimes the test case failed because of that. To make sure app registered first will help to make this more stable.